011-php-code-formatting.md

```diff
@@ -0,0 +1,94 @@
+
+StyleCI does offer to create pull requests on demand for these changes, but that workflow feels a bit disjointed given that a developer would like this all fixed in their initial pull request.
+
+It would be ideal to be able to know exactly what needs to be changed and confident that doing so has fixed the style issue; and not needing to push changes numerous times to get feedback from StyleCI on whether the changes pass or not.
+
+## Proposal
+
+We should implement the use of both the [Prettier PHP Plugin](https://github.com/prettier/plugin-php) for prettifying the look of our PHP code and the [PHP Coding Standards Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) tool for ensuring the resulting PHP code is written well, following specified code quality rules.
+
+While there is going to be some natural overlap between the two tools and how they format code style, we can definitely benefit from where the two tools differ in what they are meant to fix, to get the best of both worlds.
+
+As the [Prettier docs explain](https://prettier.io/docs/en/comparison.html), Prettier's strength lies in removing the stress of how the code "looks", and ensures it is formatted the same across a team of developers. However, it does nothing to help with code quality rules and this is where PHP CS Fixer can fill that gap, and replace the role of what StyleCI did in a much more convenient and useful way.
+
+### Prettier PHP Plugin
+
+[Prettier PHP Plugin](https://github.com/prettier/plugin-php) is an opinionated code formatter that will rewrite outputted PHP code that conforms to a consistent style, aiming for a line length print width of 80 characters.
+
+It helps developers eliminate any concerns regarding how their PHP code looks because Prettier takes care of that for them.
+
+You can see a [pull request](https://github.com/DoSomething/rogue/pull/1094) on our Rogue application with how the Prettier PHP Plugin is configured and how it proceeded to format the code.
+
+This [commit](https://github.com/DoSomething/rogue/pull/1094/commits/5e2a94b46168539a6346b6057efc2d915e444626) specifically shows how the Prettier PHP plugin is set up to run along side the configuration for our JavaScript code.
+
+This [commit](https://github.com/DoSomething/rogue/pull/1094/commits/88a8ed7d2e5f8d3e13e93c4840b2d389cd331614) specifically shows all the files changed when the `npm run format` command was executed.
+
+The way the plugin is configured, in daily use developers will not need to manually run the command to format their code since it will run as a `pre-commit` hook on any changed files.
+
+Additionally, some code editors (like Visual Studio Code) easily support running Prettier on file saves (refer to the code editor docs for documentation on setup).
```

Comment:
Sublime Text too Diego! I will not stand for this pretension and snobbery 😛

Reply:
I still ❤️  Sublime too! I use both. VSCode for large projects and ST for small and/or fast quick-and-dirty coding needs 🐎

Reply:
I love that with it running on commits and automatically on saves this isn't something I have to think about!!